edge/pkg/metamanager/metaserver/kubernetes/storage/sqlite/store.go

```diff
@@ -93,12 +95,31 @@ func (s *store) List(ctx context.Context, key string, opts storage.ListOptions,
 		return err
 	}
 	unstrList := listObj.(*unstructured.UnstructuredList)
+	_, nsFound := opts.Predicate.Field.RequiresExactMatch("metadata.namespace")
 	for _, v := range *resp.Kvs {
 		var unstrObj unstructured.Unstructured
 		err := runtime.DecodeInto(s.codec, []byte(v.Value), &unstrObj)
 		if err != nil {
 			return err
 		}
+
+		labelSet := labels.Set(unstrObj.GetLabels())
+		if !opts.Predicate.Label.Matches(labelSet) {
+			continue
+		}
+
+		// only support metadata.name & metadata.namespace
+		if nsFound && unstrObj.GetNamespace() == "" {
+			return errors.New("field label not supported: metadata.namespace")
```

Comment:
The err message above seems indicates that we don't support `metadata.namespace` field selector...

I think there's no need to return error here? If obj has no namespace, do the unstrObj.GetNamespace() will return err? If don't return err, I think we can remove the error above and just do the `!opts.Predicate.Field.Matches(fieldSet)` directly. 

If no matched obj, return the empty list, how do you think? @zc2638 @chenchunxiu

Reply:
If obj has no namespace, do the unstrObj.GetNamespace() will not return err. And I remove the error here.